### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by RudolphAschmoneit** [thingsboard/thingsboard#4225]
**Describe the bug**
I have a Device with a custom rule chain (via Device Profile), the only thing which is different from the root rule chain is a script block which gives me all values that are nested in Arrays in the Device original message as single key value pairs (I don't have the opportunity to change it in the device firmware). I get the Messages via Http/Rest ApI. At the beginning (23.02.) Everything seems to working fine, but since 05.03. I got no messages anymore, so I scroll through the logs and I found this message:

`2021-03-11 10:27:38,854 [tb-rule-engine-consumer-26-thread-3] INFO  o.t.s.s.q.DefaultTbRuleEngineConsumerService - Failed to process [1] messages`

`2021-03-11 10:27:38,855 [tb-rule-engine-consumer-26-thread-3] INFO  o.t.s.s.q.DefaultTbRuleEngineConsumerService - [ab498ec0-71c6-11eb-82dd-19d05f6ad7e0] Failed to process message: TbMsg(queueName=Main, id=b62b2198-e105-4d18-90ef-5824d32d3498, ts=1615458458845, type=POST_TELEMETRY_REQUEST, originator=020ce870-7207-11eb-82dd-19d05f6ad7e0, metaData=TbMsgMetaData(data={deviceType=AirQ, deviceName=AirQ1, ts=1615458458845}), dataType=JSON, data={"person":1,"tvoc":[1097.857,177.54],"pm2_5":[7.857,11.14],"window_open":0,"DeviceID":"5c6cd3c1b51e6e8d8e5042e3fcfc49a4","measuretime":1619.714,"cnt0_3":[1017.857,143.6],"humidity":[62.209,3.74],"window_event":0,"sound":[41.214,8.78],"Status":"OK","performance":0.0,"temperature":[17.984,0.55],"cnt0_5":[329.857,53.83],"co":[2.75,0.76],"humidity_abs":[9.583,0.45],"co2":[1667.857,100.18],"health":609.143,"uptime":41172,"cnt2_5":[1.714,10.29],"cnt10":[0.0,10.0],"so2":[1017.786,90.7],"o3":[21.519,1.62],"cnt5":[1.714,10.29],"timestamp":1615458454000,"pressure":[987.117,1.0],"TypPS":1.753,"dCO2dt":56.651,"cnt1":[47.571,19.07],"bat":[0,0],"dHdt":0.193,"oxygen":[20.868,0.81],"door_event":0.0,"pm1":[6.857,11.14],"dewpt":[10.318,0.7],"pm10":[8.429,11.3],"sound_max":59.8}, ruleChainId=5343c910-75af-11eb-8ea9-7145d5a00a69, ruleNodeId=null, ruleNodeExecCounter=0, callback=org.thingsboard.server.common.msg.queue.TbMsgCallback$1@7dd6c286), Last Rule Node: [RuleChain: AirQ Root Rule Chain|RuleNode: AirQArraysToSingleValues(4084e8a0-75b3-11eb-8ea9-7145d5a00a69)]`

The ammount of this message per day increase over time in the log files, but I also could find it on day after installation (24.02.) but only 8 times a day, now it is 4 or 5 times per second. 

I read a little bit in other issues here in the repository, ans found this one: [https://github.com/thingsboard/thingsboard/issues/887](url)

But the fixes are mentioned there (increase actors.rule.queue.max_size) did not help, I also increased ACTORS_MAX_CONCURRENT_SESSION_PER_DEVICE and 
TB_SERVER_REST_LIMITS_TENANT_CONFIGURATION but more in a try and error way, also did not help. But of course it seems like the messages are overflow the queue over time.

If I don't use the scripting Block in the Rule chain or put an output "Failed" to save timeseries I get the Messages into latest telemetry in the device and I can also see it in dashboards. Tests of the scripting block with a sample message works fine.

My Rule Chain:
![image](https://user-images.githubusercontent.com/47088627/110775771-98ae5d00-825f-11eb-8403-0b934b8c6e79.png)

My Script Function:
```var NewMsg = {
    "TypPS": msg.TypPS, 
    "bat_avg": msg.bat[0],
    "bat_err": msg.bat[1],
    "oxygen_avg": msg.oxygen[0], 
    "oxygen_err": msg.oxygen[1], 
    "pm10_avg": msg.pm10[0],
    "pm10_err": msg.pm10[1],
    "cnt0_5_avg": msg.cnt0_5[0],
    "cnt0_5_err": msg.cnt0_5[1],
    "co_avg": msg.co[0],
    "co_err": msg.co[1],
    "temperature_avg": msg.temperature[0],
    "temperature_err": msg.temperature[1],
    "performance": msg.performance, 
    "uptime": msg.uptime, 
    "window_event": msg.window_event, 
    "co2_avg": msg.co2[0],
    "co2_err": msg.co2[1],
    "so2_avg": msg.so2[0],
    "so2_err": msg.so2[1],
    "measuretime": msg.measuretime, 
    "no2_avg": msg.no2[0],
    "no2_err": msg.no2[1],
    "cnt5_avg": msg.cnt5[0],
    "cnt5_err": msg.cnt5[1],
    "dewpt_avg": msg.dewpt[0],
    "dewpt_err": msg.dewpt[1],
    "door_event": msg.door_event,
    "cnt1_avg": msg.cnt1[0],
    "cnt1_err": msg.cnt1[1],
    "pm1_avg": msg.pm1[0],
    "pm1_err": msg.pm1[1],
    "Status": msg.Status, 
    "tvoc_avg": msg.tvoc[0],
    "tvoc_err": msg.tvoc[1],
    "health": msg.health, 
    "cnt10_avg": msg.cnt10[0],
    "cnt10_err": msg.cnt10[1],
    "pressure_avg": msg.pressure[0],
    "pressure_err": msg.pressure[1],
    "dCO2dt": msg.dCO2dt, 
    "sound_max": msg.sound_max, 
    "cnt2_5_avg": msg.cnt2_5[0],
    "cnt2_5_err": msg.cnt2_5[1],
    "o3_avg": msg.o3[0],
    "o3_err": msg.o3[1],
    "humidity_avg": msg.humidity[0],
    "humidity_err": msg.humidity[1],
    "dHdt": msg.dHdt, 
    "person": msg.person, 
    "window_open": msg.window_open, 
    "humidity_abs_avg": msg.humidity_abs[0],
    "humidity_abs_err": msg.humidity_abs[1],
    "sound_avg": msg.sound[0],
    "sound_err": msg.sound[1],
    "pm2_5_avg": msg.pm2_5[0],
    "pm2_5_err": msg.pm2_5[1],
    "cnt0_3_avg": msg.cnt0_3[0],
    "cnt0_3_err": msg.cnt0_3[1]
}

NewMetadata = {
    "deviceName" : msg.DeviceID, 
    "deviceType" : "AirQ",
    "ts" : msg.timestamp
}

return {msg: NewMsg, metadata: NewMetadata, msgType: msgType};
```
A sample Message coming from Device:

```
{
  "TypPS": 1.41, 
  "bat": [0, 0], 
  "oxygen": [20.836, 0.73], 
  "pm10": [18.667, 12.42], 
  "cnt0_5": [690.5, 85.06], 
  "co": [5.311, 0.77], 
  "temperature": [28.254, 0.53], 
  "performance": 0.0, 
  "uptime": 16887, 
  "window_event": 0, 
  "co2": [2183.333, 116.38], 
  "so2": [1443.773, 117.77], 
  "DeviceID": "5c6cd3c1b51e6e8d8e5042e3fcfc49a4", 
  "measuretime": 1705.833, 
  "no2": [75.907, 3.84], 
  "cnt5": [1.667, 10.33], 
  "timestamp": 1613664328000, 
  "dewpt": [11.784, 1.2], 
  "door_event": 0.0, 
  "cnt1": [56.0, 17.51], 
  "pm1": [15.167, 11.33], 
  "Status": "OK", 
  "tvoc": [2468.167, 432.61], 
  "health": 218.833, 
  "cnt10": [0.0, 10.0], 
  "pressure": [1000.702, 1.0], 
  "dCO2dt": 106.938, 
  "sound_max": 78.80001, 
  "cnt2_5": [2.333, 10.33], 
  "o3": [2.658, 1.82], 
  "humidity": [36.87, 3.19], 
  "dHdt": 0.32, 
  "person": 2, 
  "window_open": 0, 
  "humidity_abs": [10.212, 0.75], 
  "sound": [77.75, 2.13], 
  "pm2_5": [18.333, 12.33], 
  "cnt0_3": [2165.5, 242.12]
}
```

**Your Server Environment**
* own setup
  Open Telekom Cloud, TB CE v3.2.1, Ubuntu 20.04, with external postgresql 12 (OTC managed Database) and kafka messaging queue

**Your Client Environment**
**Desktop 
* OS: macOS 10.15.7 
* Browser chrome Version 89.0.4389.82

**Your Device**

* Connectivity
  * HTTP
  
* Device vendor and model
* air-Q [https://www.air-q.com/](url)

**To Reproduce**
see descripe your bug

**Expected behavior**
That it work like in the beginning, transform all nested array values to single values and save them as timeseries

**Comment by ashvayka** [comment]
Hi @RudolphAschmoneit, try to enable debug mode in the rule node to see the exact error you experience.

**Comment by RudolphAschmoneit** [comment]
Hey, I found the Error, obviously thingsboard got problems when the original Payload is in the form {key1 : value1, key2 : value2}, so that thingsboard uses the server side ts and then I tried to overwrite it with the timestamp from the device from the message Payload before saving timeseries:

```NewMetadata = {
    "deviceName" : msg.DeviceID, 
    "deviceType" : "AirQ",
    "ts" : msg.timestamp
}

return {msg: NewMsg, metadata: NewMetadata, msgType: msgType};
```
when I changed the line "ts" : msg.timestamp back to "ts" : metadata.ts everything works fine again.
I am not really shure why it was working in the first place, If someone has any idea on this I would be very happy for sharing. Also If anyone has an idea, how I could get the timestamp of the device if I can't change the payload of the device message. Probably the answer is thingsboard Gateway or customization of the protocol, am I right?
